### ISSUE TITLE
Resource.Add doesn't add a new resource

### ISSUE THREAD

**Issue by MartinRothschink** [ical-org/ical.net#189]
This test fails in 2.2.18:

```csharp
[Test]
public void TestAddResource()
{
    var calendar = new Calendar();
    var evt = calendar.Create<Event>();

    evt.Resources.Add("Resource1");

    Assert.That(evt.Resources.Count, Is.EqualTo(1));
}
```

**Comment by rianjs** [comment]
It's a bug. `evt.Resources.UnionWith(new[]{"SomeResource"})` doesn't work either.

**Comment by MartinRothschink** [comment]
I had a quick look at the source. The getter returns a new detached hashset. Any change on the hashset is never written back into the event object:

```
        public ISet<string> Resources
        {
            get { return new HashSet<string>(Properties.GetMany<string>("RESOURCES")); }
            set { Properties.Set("RESOURCES", value); }
        }
```

**Comment by rianjs** [comment]
fbd90977305a609fbff3ced17e44cb479aae9a24

Fixed published in nuget version 2.2.26
